lib/src/shared/main/cpp/native_c4.cc

```diff
@@ -136,12 +136,18 @@ Java_com_couchbase_lite_internal_core_C4Log_getLevel(JNIEnv *env, jclass clazz,
  * Class:     com_couchbase_lite_internal_core_C4Log
  * Method:    setLevel
  * Signature: (Ljava/lang/String;I)V
+ *
+ * Since the Java code can only talk about domains that are instance of the LogDomain enum,
+ * it is ok to let this code create new domains (2nd arg to c4log_getDomain).
+ * The advantage of allowing this method to create new LogDomain instances is that if,
+ * for debugging, we need to log for a dynamically created domain, we can initialize
+ * that domain at any time, including before Core creates it.
  */
 JNIEXPORT void JNICALL
 Java_com_couchbase_lite_internal_core_C4Log_setLevel(JNIEnv *env, jclass clazz, jstring jdomain,
                                            jint jlevel) {
     jstringSlice domain(env, jdomain);
-    C4LogDomain logDomain = c4log_getDomain(domain.c_str(), false);
+    C4LogDomain logDomain = c4log_getDomain(domain.c_str(), true);
```

Comment:
Should we need to extract this to a separate JNI method?

Reply:
I thought about that for a while.  I don't see the point.  We would then have two visible methods, that customers are not allowed to use, one of which would do exactly what this one does.  It's a good idea, but I don't think it is necessary.

Reply:
this is JNI layer right, customer cannot use this directly unless we expose it via public API right?

Reply:
Well.. ya.  This is the JNI layer.  It is exactly the thing that exposes things to Java.  This function needs to be visible for use by client code (so that we can ask users to use it in their code).  It is *not* part of the public API, though, by convention, because it is in the package ...internal.core.  There are many other functions that are similarly visible, but not "public".

Reply:
Agree with Blake that I don't see a need to create a separate JNI method.